spec/features/user_contributes_new_councillor_spec.rb

```diff
@@ -18,6 +18,56 @@
       end
     end
 
+    scenario "on the contribution page" do
+      visit new_authority_councillor_contribution_path(authority.short_name_encoded)
+
+      expect(page).to have_content("Casey City Council")
+    end
+
+    context "when a person submit blank email" do
+      before :each do
+        visit new_authority_councillor_contribution_path(authority.short_name_encoded)
+
+        within ".councillor-contribution-councillors fieldset" do
+          fill_in "Full name", with: "Mila Gilic"
+          fill_in "Email", with: ""
+        end
+
+        click_button "Submit"
+      end
+
+      it "displays an error message" do
+        expect(page).to have_content("Email can't be blank")
+      end
+
+      it "does not go into the list of the suggested councillors" do
+        expect(page).to have_no_content("Full name: Mila Gilic")
+        expect(page).to have_no_content("Email:")
+      end
+    end
+
+    context "when a person submit an invalid email" do
+      before :each do
+        visit new_authority_councillor_contribution_path(authority.short_name_encoded)
+
+        within ".councillor-contribution-councillors fieldset" do
+          fill_in "Full name", with: "Mila Gilic"
+          fill_in "Email", with: "mgilic.invalid"
+        end
+
+        click_button "Submit"
+      end
+
+      it "displays an error message" do
+        expect(page).to have_content("Email format seems not right. Does it follow the format of example@email.com?")
```

Comment:
I think that while it's nice that this error message is kinda friendly saying "seems" and asking 'does it follow?', I recon error messages should be more concrete. In this case we know that the email isn't in a valid format, so we should just tell the person that and help them fix it.

What do you think of something like "Email must be a valid email address, e.g. jane@example.com" @hisayohorie ? That tells them that what they entered is not right, what the requirement is to get it accepted, and keeps the helpful example you added, but human name in it to make it a little more human 🌳 . That's also a bit more succinct.